main.tf

```diff
@@ -46,7 +46,7 @@ resource "local_file" "private_key_pem" {
 }
 
 resource "null_resource" "chmod" {
-  count      = "${var.generate_ssh_key == "true" ? 1 : 0}"
+  count      = "${var.generate_ssh_key == "true" && var.operating_system == "unix_like" ? 1 : 0}"
   depends_on = ["local_file.private_key_pem"]
 
   provisioner "local-exec" {
```

Comment:
@aknysh do you have any other suggestions on how to solve this?

I'm thinking rather than introduce this `operating_system` boolean, that we parameterize the command. E.g. `chmod_command="chmod 600"`

If the user sets `chmod_command=""`, then it won't run. Also, then on other systems, the user has the ability to modify the command executed. I like this solution because it provides additional capabilities. Also, the `operating_system` flag is a little bit confusing because it refers to the "operator's operating system", but not the remote operating system.

Reply:
@osterman 
I like your approach of having a parameterized command.

The only challenge here is the file name in the final `command`

```hcl
command = "chmod 600 ${local.private_key_filename}"
```
 
Interpolation is not allowed in variables, so it would look like

```hcl
command = "${var.chmod_command}  ${local.private_key_filename}"
```

Since `${local.private_key_filename}` is always here, `var.chmod_command` works like a boolean flag (a switch b/w Linux/Windows ) and not like a placeholder for any possible command.

How do you think we could improve on the idea?

Reply:
Although we can make `${local.private_key_filename}` empty if var.chmod_command = ""

Reply:
Aha, good point. So what we want is `command` to be a template. So it would be rather:
```
command = "${format(var.chmod_command, local.private_key_filename)}" 
```

Then:
```
variable "chmod_command" {
  default = "chmod 600 %v"
}
```